Array & Hash/kadane_algorithm.py

```diff
@@ -1,16 +1,22 @@
 # Problem link: https://leetcode.com/problems/maximum-subarray/
-# Best solution explanation: https://www.youtube.com/watch?v=5WZl3MMT0Eg
+#  solution explanation: https://www.youtube.com/watch?v=AHZpyENo7k4&t=939s
+
 
 # Solution 1:
 # Complexity: O(n)
 
 def maxSubArray(nums):
     maxSum = nums[0]
-    currentSum = 0
-    for n in nums:
-        if currentSum < 0:
-            currentSum = 0
-        currentSum += n
-        maxSum = max(maxSum, currentSum)
+    sum = 0
+    for item in nums:
+        if sum < 0:
+            sum = 0
+        sum += item
+        if sum>maxSum:
+            maxSum=sum
+        
```

Comment:
Try some different approach, it is exactly similar to the previous one. Moreover don't remove the previously added code, add your solution with # Solution 2 heading 
Or either prefer a different language